packages/backend/src/open_payments/receiver/model.ts

```diff
@@ -74,39 +52,31 @@ export class Receiver extends ConnectionBase {
       : undefined
     const receivedAmount = parseAmount(incomingPayment.receivedAmount)
 
-    if (!isValidIlpAddress(incomingPayment.ilpStreamConnection.ilpAddress)) {
-      throw new Error('Invalid ILP address on stream connection')
-    }
-
-    return new this(
-      {
-        ...incomingPayment.ilpStreamConnection,
-        ilpAddress: incomingPayment.ilpStreamConnection.ilpAddress
-      },
-      {
-        id: incomingPayment.id,
-        completed: incomingPayment.completed,
-        walletAddress: incomingPayment.walletAddress,
-        expiresAt,
-        receivedAmount,
-        incomingAmount,
-        metadata: incomingPayment.metadata,
-        createdAt: new Date(incomingPayment.createdAt),
-        updatedAt: new Date(incomingPayment.updatedAt)
-      }
+    // TODO: handle multiple payment methods
+    const ilpMethod = incomingPayment.methods.find(
+      (method) => method.type === 'ilp'
     )
-  }
+    if (!ilpMethod) {
+      throw new Error('Cannot create receiver from unsupported payment method')
+    }
+    if (!isValidIlpAddress(ilpMethod.ilpAddress)) {
+      throw new Error('Invalid ILP address on ilp payment method')
+    }
 
-  private constructor(
-    connection: OpenPaymentsConnectionWithIlpAddress,
-    public incomingPayment?: ReceiverIncomingPayment
-  ) {
-    super(
-      connection.ilpAddress,
-      base64url.toBuffer(connection.sharedSecret),
-      connection.assetCode,
-      connection.assetScale
-    )
+    this.ilpAddress = ilpMethod.ilpAddress
+    this.sharedSecret = base64url.toBuffer(ilpMethod.sharedSecret)
+    // TODO: ensure these are the right assetCode/Scales
+    this.assetCode = incomingPayment.receivedAmount.assetCode
+    this.assetScale = incomingPayment.receivedAmount.assetScale
```

Comment:
I will address this TODO before merging. Previously we were getting these off the connection and I wasn't sure where we should get it now - Max suggested this and I believe it is correct. Tests seem to back it up but any more common-sense checks would be welcome https://github.com/interledger/rafiki/blob/04051d144787eb0bee75283416ccd6002f07d3f2/packages/backend/src/open_payments/receiver/model.test.ts#L49-L51

Reply:
Other than fetching the walletAddress asset, I don't think there is really any other option that we have